### PR TITLE
Janitorial: bump various dependencies

### PR DESCRIPTION
**Pull request by tbradsha**
This bumps a few dependencies now that we're no longer bound to PHP <7.2.

## Proposed changes:

The details are in each commit, but mostly the bumps didn't cause any issues, with one exception that required some minor adjustment due to a PhpDocParser API change: 78c163b4436af0b4c8e25b7f155117bce52d094d

I also updated all packages using `automattic/wordbless` to `^0.4.2` instead of variations that ultimately pulled the same thing.
<details>
<summary>Other dependencies that could be bumped</summary>

### yoast/phpunit-polyfills
Currently there are 90 packages on `^1.1.1`. This is a big project (see also p1732198254207959/1732037783.177369-slack-C05Q5HSS013).

### symfony/process
This is found in `automattic/jetpack-changelogger`. 
Currently first three items have (Windows) security bulletins. `5.4.46` is the first without one, but requires PHP 7.2.5, so we may as well bump PHP too, then bump this dependency to `^5.4 || ^6.4 || ^7.1`

### symfony/console
This is found in `automattic/jetpack-changelogger`. I recommend bumping the dependency to `^5.4 || ^6.4 || ^7.1` (assuming we'll be bumping PHP to `7.2.5`; see above).

### dompdf/dompdf
This is found in `automattic/jetpackcrm`. I tried bumping it to `^3.0`, but this messed up formatting, so will take a bit more digging.

### composer-plugin-api and composer/composer
Both `automattic/jetpack-autoloader` and `automattic/jetpack-composer-plugin` use these. I believe they can bumped safely (and/or removed) but documentation is scarce as to why we use them in the first place.

### dealerdirect/phpcodesniffer-composer-installer
Both `automattic/jetpack-codesniffer` and `automattic/wpcomsh` use this. This one is over my head as well.

</details>

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Verify tests pass?

### CODE REVIEW

File: projects/packages/patchwork-redefine-exit/composer.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	"license": "GPL-2.0-or-later",
 	"require": {
 		"php": ">=7.2",
-		"antecedent/patchwork": "^2.1"
+		"antecedent/patchwork": "^2.2"
```
Comment (anomiex) [inline projects/packages/patchwork-redefine-exit/composer.json:15]: I note this change doesn't really make a difference, since `^2.1` allows 2.2 as well.

**Review by anomiex — APPROVED** [review]
LGTM. Although, did you intentionally skip changelogger's symfony deps?

File: projects/packages/patchwork-redefine-exit/composer.json (hunk 1 above)
Comment (tbradsha) [reply projects/packages/patchwork-redefine-exit/composer.json:15]: Yeah, it's more of a philosophical difference; 2.1 is for PHP >=5.4 and 2.2 requires at least PHP 7.1, so I wanted to remove the mismatch.

**Comment by tbradsha** [comment]
> Although, did you intentionally skip changelogger's symfony deps?

Heh, there's a reason this was a draft. I added some details in the `Other dependencies that could be bumped` spoiler. More immediately relevant, though, I was waiting for #40162 to be merged so I didn't have to deal with any conflicts from that.

**Comment by anomiex** [comment]
> composer-plugin-api and composer/composer
>
> Both automattic/jetpack-autoloader and automattic/jetpack-composer-plugin use these. I believe they can bumped safely (and/or removed) but documentation is scarce as to why we use them in the first place.

composer/composer is in there for phan, IDE completions, and possibly also unit tests IIRC. Should be safe to bump.

composer-plugin-api indicates the Composer API version the package is compatible with, in much the same way that the `"php": ">=7.2"` indicates which version of PHP the package is compatible with. See https://getcomposer.org/doc/articles/plugins.md#plugin-package for some more details.

> dealerdirect/phpcodesniffer-composer-installer
>
> Both automattic/jetpack-codesniffer and automattic/wpcomsh use this. This one is over my head as well.

Should be safe to bump. Probably wpcomsh doesn't even need it (or wp-coding-standards/wpcs) anymore now that it's in the monorepo.

What this does is, when a phpcs standards package like wp-coding-standards/wpcs is installed, it automatically adds the right phpcs configuration so phpcs knows about it.

**Comment by tbradsha** [comment]
Great! I'll tackle those in another PR.

Edit: #40297

**Pull request merged**